pandas/_libs/groupby_helper.pxi.in

```diff
@@ -513,26 +514,40 @@ def group_rank_{{name}}(ndarray[float64_t, ndim=2] out,
             # Used to calculate tiebreakers
             dups += 1
             sum_ranks += i - grp_start + 1
-
             # if keep_na, check for missing values and assign back
             # to the result where appropriate
-
             if keep_na and mask[_as[i]]:
                 grp_na_count += 1
                 out[_as[i], 0] = nan
-            else:
-                # this implementation is inefficient because it will
-                # continue overwriting previously encountered dups
-                # i.e. if 5 duplicated values are encountered it will
-                # write to the result as follows (assumes avg tiebreaker):
-                # 1
-                # .5  .5
-                # .33 .33 .33
-                # .25 .25 .25 .25
-                # .2  .2  .2  .2  .2
-                #
-                # could potentially be optimized to only write to the
-                # result once the last duplicate value is encountered
+                grp_vals_seen = i + 1
+                # when label transition happens, update grp_size to keep track
+                # of number of nans encountered and increment grp_tie_count
+                if (i== N-1) or (labels[_as[i]] != labels[_as[i+1]]):
```

Comment:
Does this section pass LINTing? Whitespace between operators and operands looks off

Reply:
usually I use ``flake8 <python-filepath>``. (I am using a Windows machine) But since it is not a standard .py file. There are too many pep8 issues listing. I have to find out the real issue by examining each of them. Any suggestions on a better way to deal with it ?

Reply:
Hmm ok that’s right I suppose these files are not flakeable. I suppose just make sure to have whitespace  on both sides of the relational operators